addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHttpMethodScanRule.java

```diff
@@ -642,4 +622,80 @@ private void testHttpMethod(String httpMethod) throws Exception {
     private static String randomAlphanumeric(int count) {
         return RandomStringUtils.secure().nextAlphanumeric(count);
     }
+
+    private AlertBuilder buildAlert(
+            String vulnName,
+            String vulnDesc,
+            String extraInfo,
+            String evidence,
+            HttpMessage msg,
+            VulnType currentVT) {
+        return newAlert()
+                .setConfidence(Alert.CONFIDENCE_MEDIUM)
+                .setName(
+                        Constant.messages.getString(
+                                "ascanbeta.insecurehttpmethod.detailed.name", vulnName))
+                .setDescription(vulnDesc)
+                .setOtherInfo(extraInfo)
+                .setSolution(Constant.messages.getString("ascanbeta.insecurehttpmethod.soln"))
+                .setEvidence(evidence)
+                .setMessage(msg)
+                .setCweId(getCweId())
+                .setWascId(getWascId())
+                .setAlertRef(getId() + "-" + currentVT.getRef());
+    }
+
+    @Override
+    public List<Alert> getExampleAlerts() {
+        List<Alert> alerts = new ArrayList<>();
+
+        for (VulnType vulnType : VulnType.values()) {
+            String name = "";
+            String description = "";
+            String extraInfo = "";
+            String evidence = "";
+
+            switch (vulnType) {
+                case DELETE_METHOD_ENABLED:
+                    name = "DELETE Method Enabled";
+                    description = "The server allows the DELETE HTTP method which can be unsafe.";
+                    extraInfo = "DELETE requests could remove resources if exploited.";
+                    evidence = "DELETE";
```

Comment:
These values should be the actual values used in the alerts, though you may need to provide the details or specifics like "DELETE". You could also include these int he enum members as that likely simplifies things. It also ensure that they're translated etc.